pkg/controllers/cassandra/cassandra.go

```diff
@@ -48,15 +51,22 @@ func NewCassandra(
 	)
 
 	cc := &CassandraController{
-		queue: queue,
+		queue:    queue,
+		recorder: recorder,
 	}
 	cassClusters.AddEventHandler(&controllers.QueuingEventHandler{Queue: queue})
 	cc.cassLister = listersv1alpha1.NewCassandraClusterLister(
 		cassClusters.GetIndexer(),
 	)
 	cc.cassListerSynced = cassClusters.HasSynced
-
-	cc.control = NewControl(recorder)
+	cc.control = NewControl(
+		service.NewControl(
+			kubeClient,
+			corev1listers.NewServiceLister(services.GetIndexer()),
+			recorder,
+		),
+		recorder,
+	)
```

Comment:
You've not stored a reference to the `HasSynced` function of the `services` SharedIndexInformer, so this controller will start *before* the service lister has synced. It should be done similar to how you've done with the `cassClusters` SharedIndexInformer.

Reply:
Done.